### DISCUSSION TITLE
How popular does a language need to be for a parser to be merged in?

### DISCUSSION THREAD

**Discussion by alexkar598** [tree-sitter/tree-sitter#1143] [Q&A] [answered]
I'm considering writing a parser for [DM](https://secure.byond.com/docs/ref/) which is a custom language made for a rather old but still maintained 2D game engine/game service called [BYOND](http://www.byond.com/). It is by no means a popular language but its still actively used, a good chunk of these users using it for the development of [Space Station 13](https://github.com/topics/ss13). Since we use github to store our repos, having language support for DM (when github eventually updates dependencies) would be great.

My question is would such a parser if written be accepted into the repo or would it be denied on the basis of being too niche to support? I'd rather not spend time writing a parser if it has no chance of getting into the main repo.

**Comment by ahlinc** [comment, marked as answer]
> My question is would such a parser if written be accepted into the repo or would it be denied on the basis of being too niche to support? I'd rather not spend time writing a parser if it has no chance of getting into the main repo.

There are no language parsers in this tree-sitter main repo, you only may ask to mention your parser with a link to your repo on the project's [main page](https://tree-sitter.github.io/tree-sitter/#available-parsers) by opening a PR against this repo. That's what many people doing and if you check the mentioned page you'll see that nearly half of languages leaves in authors' repos outside of tree-sitter  Github organization.

**Reply by alexkar598** [reply]
That explains alot, thank you
